src/commands/mod.rs

```diff
@@ -580,13 +581,25 @@ impl DaemonControl {
         }
     }
 
-    pub fn list_spend(&self) -> ListSpendResult {
+    pub fn list_spend(&self, txids: Option<Vec<bitcoin::Txid>>) -> ListSpendResult {
         let mut db_conn = self.db.connection();
-        let spend_txs = db_conn
-            .list_spend()
-            .into_iter()
-            .map(|(psbt, updated_at)| ListSpendEntry { psbt, updated_at })
-            .collect();
+        let spend_psbts = db_conn.list_spend();
+        let mut spend_txs: Vec<ListSpendEntry> = Vec::new();
+        for (psbt, updated_at) in spend_psbts {
+            let tx: &Transaction = &psbt.unsigned_tx;
+            if let Some(txids_list) = txids.as_ref() {
+                for &txid in txids_list {
+                    if tx.txid() == txid {
+                        spend_txs.push(ListSpendEntry {
+                            psbt: psbt.clone(),
+                            updated_at,
+                        });
+                    }
+                }
+            } else {
+                spend_txs.push(ListSpendEntry { psbt, updated_at });
+            };
```

Comment:
Meh at the `O(N * M)` complexity here. I'd rather create a `HashSet` from the `Vector` of txids to be able in the loop to check presence from the set in `O(1)`. (That effectively makes it `O(N) + O(M)` instead.)

Side nit: your can also use `filter_map`.